src/Groups/GAPGroups.jl

```diff
@@ -588,13 +649,13 @@ Return whether generators for the group `G` are known.
 # Examples
 ```jldoctest
 julia> F = free_group(2)
-Free group of rank 2
+Free group of rank 2 with 2 generators [f1, f2]
 
 julia> has_gens(F)
 true
 
 julia> H = derived_subgroup(F)[1]
-Free group
+Free group of unknown rank
```

Comment:
For free groups we already printed the rank if it was known, but nothing if it was unknown. Now I print explicitly "of unknown rank". Why is visible in this example:
```
julia> f = free_group(4)
Free group of rank 4 with 4 generators [f1, f2, f3, f4]

julia> h = sub(f, [f[1],f[3], f[3]*f[1]])[1]
Free group of unknown rank with 3 generators [f1, f3, f3*f1]
```
So this subgroup is a free group of rank 2, but it was printed as `Free group with 3 generators` which I think one can very very easily mistake for implying that the rank is 3.

Or perhaps it is just me who is not careful enough... but I really think this is beneficial, and the downsides seem to be minimal?